sdk/src/main/scala/com/horizen/account/api/http/AccountWalletApiRoute.scala

```diff
@@ -0,0 +1,152 @@
+package com.horizen.account.api.http
+
+import akka.actor.{ActorRef, ActorRefFactory}
+import akka.http.scaladsl.server.Route
+import akka.pattern.ask
+import com.fasterxml.jackson.annotation.JsonView
+import com.horizen.account.api.http.AccountWalletErrorResponse.ErrorCouldNotGetBalance
+import com.horizen.account.api.http.AccountWalletRestScheme.{ReqGetBalance, ReqGetTotalBalance, RespCreatePrivateKeySecp256k1, RespGetBalance}
+import com.horizen.{AbstractSidechainNodeViewHolder, SidechainTypes}
+import com.horizen.account.block.{AccountBlock, AccountBlockHeader}
+import com.horizen.account.chain.AccountFeePaymentsInfo
+import com.horizen.account.node.{AccountNodeView, NodeAccountHistory, NodeAccountMemoryPool, NodeAccountState}
+import com.horizen.account.proposition.AddressProposition
+import com.horizen.account.secret.{PrivateKeySecp256k1, PrivateKeySecp256k1Creator}
+import com.horizen.api.http.WalletBaseErrorResponse.ErrorSecretNotAdded
+import com.horizen.api.http.{ApiResponseUtil, ErrorResponse, SuccessResponse, WalletBaseApiRoute}
+import com.horizen.node.NodeWalletBase
+import com.horizen.proposition.Proposition
+import com.horizen.serialization.Views
+import com.horizen.utils.BytesUtils
+import sparkz.core.settings.RESTApiSettings
+
+import java.util.{Optional => JOptional}
+import scala.concurrent.{Await, ExecutionContext}
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+import com.horizen.api.http.JacksonSupport._
+import com.horizen.api.http.WalletBaseRestScheme.ReqCreateKey
+
+import java.math.BigInteger
+import scala.collection.JavaConverters.asScalaBufferConverter
+
+
+case class AccountWalletApiRoute(override val settings: RESTApiSettings,
+                                 sidechainNodeViewHolderRef: ActorRef)(implicit override val context: ActorRefFactory, override val ec: ExecutionContext)
+  extends WalletBaseApiRoute[
+    SidechainTypes#SCAT,
+    AccountBlockHeader,
+    AccountBlock,
+    AccountFeePaymentsInfo,
+    NodeAccountHistory,
+    NodeAccountState,
+    NodeWalletBase,
+    NodeAccountMemoryPool,
+    AccountNodeView] (settings, sidechainNodeViewHolderRef) {
+
+  override val route: Route = pathPrefix("wallet") {
+    // some of these methods are in the base class
```

Comment:
UTXO methods: `importSecret, exportSecret, dumpSecrets, importSecrets` could be invoked in Account context too if moved in common base class